### DISCUSSION TITLE
How to make a change character event?

### DISCUSSION THREAD

**Discussion by m4ttzzzz** [CodenameCrew/CodenameEngine#458] [Q&A]
Hi. How to make a change character event? I was looking through the Wiki and the API, but couldn't find such thing. (im not smart with hscript 🙏🙏, also the closest thing i found is the Characters array in StrumLine)

Edit: How do I also change the stage?

**Comment by r6915ee** [comment]
> Hi. How to make a change character event? I was looking through the Wiki and the API, but couldn't find such thing. (im not smart with hscript 🙏🙏, also the closest thing i found is the Characters array in StrumLine)

At least, you would need to change the character, the icon, and the color, though I do not exactly know if anything else needs to be done as well. The characters array is also a bit far from what you want, because all characters use the same "anchor," per say, at least that's what I like to call it: essentially the origin for anything attached to it. At most, the characters array is for multiple characters (though it's not really the best tool for it as you would need to create additional note types), though you could still use it for other purposes, such as a different perspective.

> How do I also change the stage?

If that stage only gets used once, then you can define it along with your first stage in the same files, and set the correct properties once everything has been loaded up. Afterwards, at certain points in time (e.g. using steps), you can make the new stage appear.

**Reply by m4ttzzzz** [reply]
Thanks for answering! I get it now. :-)

**Comment by THEkatinamicrowave** [comment]
Based on how Codename works, my guess would be

1. Make an event HX and JSON with the fields "character array" "character number" "new char" and "update?"
2. locate the character (character number) in the array (character array) and then swap it out for a new character (new char) using the funkin.game.Character
3. if update (update?), update the health bar and color

I'm kinda interested in this myself so I might try doing this
I know that Impostor!Mania mod has a change character script; if I were you I'd just yoink it

Edit: found this thing in PlayState.hx. Make do.
![image](https://github.com/user-attachments/assets/4093d7b2-90d1-4ec6-9d82-c9b509a5db9d)
